### ISSUE TITLE
Configuration Manager dont start

### ISSUE THREAD

**Issue by remnts** [tacacsgui/tacacsgui#51]
**Describe the system**
Ubuntu version: ... 
(use next command: lsb_release -a )
PHP version: ... 
(use next command: php -v )
TacacsGUI API version: **IMPORTANT** 
(use next command: php -r 'include "/opt/tacacsgui/web/api/constants.php"; echo APIVER . "\n";' )
Browser: [e.g. chrome, safari]

**Describe the bug**

**To Reproduce**
3. See error

**Screenshots**
Image
![imagen](https://user-images.githubusercontent.com/328660/61308033-9e168400-a7ac-11e9-9a49-e61fed5abe8d.png)

**Additional context**
In the las GUI version i have this error whem clic on star button: 

Server Error Appeared!
Description:

Type: _undefined_: _undefined_

Status: 704

Status Text: Internal Server Error

Response Text: { "message": "Slim Application Error", "exception": [ { "type": "Exception", "code": 0, "message": "Traceback (most recent call last):\n File \"\/opt\/tacacsgui\/plugins\/ConfigManager\/cm.py\", line 47, in \n hours, minutes = cm.get('time', '00:00').split(':')\nValueError: not enough values to unpack (expected 2, got 1)", "file": "\/opt\/tacacsgui\/web\/api\/app\/Services\/CMDRun\/CMDRun.php", "line": 80, "trace": [ "#0 \/opt\/tacacsgui\/web\/api\/app\/Controllers\/ConfManager\/ConfManager.php(1113): tgui\\Services\\CMDRun\\CMDRun->get()", "#1 \/opt\/tacacsgui\/web\/api\/app\/Controllers\/ConfManager\/ConfManager.php(716): tgui\\Controllers\\ConfManager\\ConfManager->startCron(Array)", "#2 [internal function]: tgui\\Controllers\\ConfManager\\ConfManager->postCron(Object(Slim\\Http\\Request), Object(Slim\\Http\\Response), Array)", "#3 \/opt\/tacacsgui\/web\/api\/vendor\/slim\/slim\/Slim\/Handlers\/Strategies\/RequestResponse.php(41): call_user_func(Array, Object(Slim\\Http\\Request), Object(Slim\\Http\\Response), Array)", "#4 \/opt\/tacacsgui\/web\/api\/vendor\/slim\/slim\/Slim\/Route.php(356): Slim\\Handlers\\Strategies\\RequestResponse->__invoke(Array, Object(Slim\\Http\\Request), Object(Slim\\Http\\Response), Array)", "#5 \/opt\/tacacsgui\/web\/api\/vendor\/slim\/slim\/Slim\/MiddlewareAwareTrait.php(117): Slim\\Route->__invoke(Object(Slim\\Http\\Request), Object(Slim\\Http\\Response))", "#6 \/opt\/tacacsgui\/web\/api\/vendor\/slim\/slim\/Slim\/Route.php(334): Slim\\Route->callMiddlewareStack(Object(Slim\\Http\\Request), Object(Slim\\Http\\Response))", "#7 \/opt\/tacacsgui\/web\/api\/vendor\/slim\/slim\/Slim\/App.php(516): Slim\\Route->run(Object(Slim\\Http\\Request), Object(Slim\\Http\\Response))", "#8 \/opt\/tacacsgui\/web\/api\/app\/Middleware\/ChangeHeaderMiddleware.php(14): Slim\\App->__invoke(Object(Slim\\Http\\Request), Object(Slim\\Http\\Response))", "#9 [internal function]: tgui\\Middleware\\ChangeHeaderMiddleware->__invoke(Object(Slim\\Http\\Request), Object(Slim\\Http\\Response), Object(Slim\\App))", "#10 \/opt\/tacacsgui\/web\/api\/vendor\/slim\/slim\/Slim\/DeferredCallable.php(43): call_user_func_array(Object(tgui\\Middleware\\ChangeHeaderMiddleware), Array)", "#11 [internal function]: Slim\\DeferredCallable->__invoke(Object(Slim\\Http\\Request), Object(Slim\\Http\\Response), Object(Slim\\App))", "#12 \/opt\/tacacsgui\/web\/api\/vendor\/slim\/slim\/Slim\/MiddlewareAwareTrait.php(70): call_user_func(Object(Slim\\DeferredCallable), Object(Slim\\Http\\Request), Object(Slim\\Http\\Response), Object(Slim\\App))", "#13 \/opt\/tacacsgui\/web\/api\/vendor\/slim\/slim\/Slim\/MiddlewareAwareTrait.php(117): Slim\\App->Slim\\{closure}(Object(Slim\\Http\\Request), Object(Slim\\Http\\Response))", "#14 \/opt\/tacacsgui\/web\/api\/vendor\/slim\/slim\/Slim\/App.php(407): Slim\\App->callMiddlewareStack(Object(Slim\\Http\\Request), Object(Slim\\Http\\Response))", "#15 \/opt\/tacacsgui\/web\/api\/vendor\/slim\/slim\/Slim\/App.php(315): Slim\\App->process(Object(Slim\\Http\\Request), Object(Slim\\Http\\Response))", "#16 \/opt\/tacacsgui\/web\/api\/index.php(5): Slim\\App->run()", "#17 {main}" ] } ] }

message: ===>Slim Application Error<===(string)

exception: '0' : 'type' => "Exception" 'code' => "0" 'message' => "Traceback (most recent call last): File "/opt/tacacsgui/plugins/ConfigManager/cm.py", line 47, in hours, minutes = cm.get('time', '00:00').split(':') ValueError: not enough values to unpack (expected 2, got 1)" 'file' => "/opt/tacacsgui/web/api/app/Services/CMDRun/CMDRun.php" 'line' => "80" 'trace' : '0' => "#0 /opt/tacacsgui/web/api/app/Controllers/ConfManager/ConfManager.php(1113): tgui\Services\CMDRun\CMDRun->get()" '1' => "#1 /opt/tacacsgui/web/api/app/Controllers/ConfManager/ConfManager.php(716): tgui\Controllers\ConfManager\ConfManager->startCron(Array)" '2' => "#2 [internal function]: tgui\Controllers\ConfManager\ConfManager->postCron(Object(Slim\Http\Request), Object(Slim\Http\Response), Array)" '3' => "#3 /opt/tacacsgui/web/api/vendor/slim/slim/Slim/Handlers/Strategies/RequestResponse.php(41): call_user_func(Array, Object(Slim\Http\Request), Object(Slim\Http\Response), Array)" '4' => "#4 /opt/tacacsgui/web/api/vendor/slim/slim/Slim/Route.php(356): Slim\Handlers\Strategies\RequestResponse->__invoke(Array, Object(Slim\Http\Request), Object(Slim\Http\Response), Array)" '5' => "#5 /opt/tacacsgui/web/api/vendor/slim/slim/Slim/MiddlewareAwareTrait.php(117): Slim\Route->__invoke(Object(Slim\Http\Request), Object(Slim\Http\Response))" '6' => "#6 /opt/tacacsgui/web/api/vendor/slim/slim/Slim/Route.php(334): Slim\Route->callMiddlewareStack(Object(Slim\Http\Request), Object(Slim\Http\Response))" '7' => "#7 /opt/tacacsgui/web/api/vendor/slim/slim/Slim/App.php(516): Slim\Route->run(Object(Slim\Http\Request), Object(Slim\Http\Response))" '8' => "#8 /opt/tacacsgui/web/api/app/Middleware/ChangeHeaderMiddleware.php(14): Slim\App->__invoke(Object(Slim\Http\Request), Object(Slim\Http\Response))" '9' => "#9 [internal function]: tgui\Middleware\ChangeHeaderMiddleware->__invoke(Object(Slim\Http\Request), Object(Slim\Http\Response), Object(Slim\App))" '10' => "#10 /opt/tacacsgui/web/api/vendor/slim/slim/Slim/DeferredCallable.php(43): call_user_func_array(Object(tgui\Middleware\ChangeHeaderMiddleware), Array)" '11' => "#11 [internal function]: Slim\DeferredCallable->__invoke(Object(Slim\Http\Request), Object(Slim\Http\Response), Object(Slim\App))" '12' => "#12 /opt/tacacsgui/web/api/vendor/slim/slim/Slim/MiddlewareAwareTrait.php(70): call_user_func(Object(Slim\DeferredCallable), Object(Slim\Http\Request), Object(Slim\Http\Response), Object(Slim\App))" '13' => "#13 /opt/tacacsgui/web/api/vendor/slim/slim/Slim/MiddlewareAwareTrait.php(117): Slim\App->Slim\{closure}(Object(Slim\Http\Request), Object(Slim\Http\Response))" '14' => "#14 /opt/tacacsgui/web/api/vendor/slim/slim/Slim/App.php(407): Slim\App->callMiddlewareStack(Object(Slim\Http\Request), Object(Slim\Http\Response))" '15' => "#15 /opt/tacacsgui/web/api/vendor/slim/slim/Slim/App.php(315): Slim\App->process(Object(Slim\Http\Request), Object(Slim\Http\Response))" '16' => "#16 /opt/tacacsgui/web/api/index.php(5): Slim\App->run()" '17' => "#17 {main}"

url: ===>/api/confmanager/settings/cron/<===(string)

data: 'cm' : 'period' => "day" 'time' => "" 'week' => "1" 'git' : 'period' => "60"

Debug info:

    PHP version: 7.3.3-1
    API version: 0.9.63
    tac_plus version: 201903091339/PCRE/LWRES

**Comment by gurkin33** [comment]
Hello,

I think you have to try [re-install GUI](https://tacacsgui.com/documentation/installation/re-installation/).

Regards, Aleksey

**Comment by Alex-0293** [comment]
Hello,
I have the same problem. I'm reinstalling GUI, first time it`s work fine.

**Comment by gurkin33** [comment]
Hello,

Is it worked before or is it first try?

Regards, Aleksey

**Comment by Alex-0293** [comment]
It did not work before.

**Comment by Alex-0293** [comment]
What logs are useful, to understand this problem?

**Comment by svenron** [comment]
I had the exact same issue.
When I picked another time like 01:05 it worked.
Console was complaining for invalid day in crontab.
So the minutes can't be 0. There is an issue.

**Comment by kpt-smirk** [comment]
The problem is either 
- the regex replacement in cm.py (/opt/tacacsgui/plugins/ConfigManager/pm.py line 54++ )
- or that /opt/tgui_data/confManager/cron.yaml doesn't have the time in double digits like "00:00"

If removed the regex in cm.py in the cron-set part and it works for me now.

hope it helps 

```
--- cm.py	2020-02-23 13:05:49.440062457 +0100
+++ cm.py.orig	2020-02-23 12:59:34.510130584 +0100
@@ -47,11 +47,11 @@
     hours, minutes = cm.get('time', '00:00').split(':')
     import re
     cron_content ='''#### TGUI CM ####
-{} {} * * {} {} -c /opt/tgui_data/confManager/config.yaml > /dev/null 2>/dev/null &
+{} {}  * * {} {} -c /opt/tgui_data/confManager/config.yaml > /dev/null 2>/dev/null &
 #### TGUI GIT ####
 */{} * * * * {} --git-commit -c /opt/tgui_data/confManager/config.yaml > /dev/null 2>/dev/null &
 '''.format(
-        minutes, hours, week,
+        re.sub('^0', '', minutes), re.sub('^0', '', hours), week,
         os.path.realpath(__file__), git.get('period', 60), os.path.realpath(__file__)
     )
     try:

```

**Comment by willcurtis** [comment]
Hi Guys, i am also having the same issue with the configuration manager, I have reinstalled and tried the above solution from @kpt-smirk to no avail.

@gurkin33 I'm happy to pay the $60 for support if the project is still being maintained? I would like some reassurances it isnt going to dissappear in the next 12 months though :)

**Comment by gurkin33** [comment]
Dear @willcurtis ,
Configuration Manger is a plugin, it is not main part of TACACSGUI. I hope TACACSGUI as a service which manages tacacs daemon works for you. We are preparing big update for TACACSGUI and the main idea right now is to switch from PHP to Python with adding some useful new features. And if you like this project, like what we already done as a main idea of TACACSGUI and you want to support us you can do this any time.
Thank you.
Regards, Aleksey

**Comment by willcurtis** [comment]
Thanks, are you not able to assist with the configuration manager then?

**Comment by gurkin33** [comment]
Better say we see how we can rebuild it, but doesn't have enough time and finally it will have another architecture in a new version, that why it will be a double work.
Sorry.

**Comment by gorgeousyouth** [comment]
excuse me, when the Configuration Manager start the Configuration Manager log reports the error：
Timeout exceeded.
<pexpect.pty_spawn.spawn object at 0x7f36c3b6d9e8>
command: /usr/bin/telnet
args: [b'/usr/bin/telnet', b' ****', b'***** ']
buffer (last 100 chars): '        *\r\n******************************************************************************\r\n\r\nlogin: '
before (last 100 chars): '        *\r\n******************************************************************************\r\n\r\nlogin: '
after: <class 'pexpect.exceptions.TIMEOUT'>
match: None
match_index: None
exitstatus: None
flag_eof: False
pid: 8770
child_fd: 6
closed: False
timeout: 15
delimiter: <class 'pexpect.exceptions.EOF'>
logfile: None
logfile_read: None
logfile_send: None
maxread: 2000
ignorecase: False
searchwindowsize: None
delaybeforesend: 0.05
delayafterclose: 0.1
delayafterterminate: 0.1
searcher: searcher_re:
    0: re.compile('login as:')
    1: re.compile('sername:')
I think the path caused this error, however , i can not find .would you mind giving me the resolution?

**Comment by gurkin33** [comment]
Hello,
I think you set incorrect prompt 'login as:' and 'sername:', but as I see login prompt for you is 'login:'
Could you try this?

**Comment by sanjeevtelexair** [comment]
I had configured the Configuration manager by configuring in the below sequence:
1. Credentials
2. Models
3. Devices
4. Queries
The configuration manager starts and the backup also successful.

However, when I configure more than one device in one query or one query for each device, it is taking backup of only the first device and only the first query. It is not taking the backup of 2nd device or 2nd query!
Dear Aleksey - Can you please throw some light?

Regards'
Sanjeeva Reddy

**Comment by tmcruisin** [comment]
Hi All,

Old thread but I was stuck with this issue - Would only do the first Device in the first Query (i.e. first entry in config.yaml). We resolved the issue by commenting out lines 211 & 214 in cm.py. We also had to use yaml.safe_load for another issue.

```
+ cm.py.working
-  cm.py.orig

116c116
+             data_loaded = yaml.safe_load(stream)
-             data_loaded = yaml.load(stream)
122c122
+             data_loaded = yaml.safe_load(stream)
-             data_loaded = yaml.load(stream)
211c211
+                       # loggin.add(**log_data, status='success')
-                       loggin.add(**log_data, status='success')
214c214
+               # if not args.test_queries: loggin.add(**log_data, status='error', message=e)
-               if not args.test_queries: loggin.add(**log_data, status='error',  #message=e)
``` 

HTH.

Cheers
tmcruisin [tm]
